Sources/NIOSSL/NIOSSLHandler.swift

```diff
@@ -552,6 +613,54 @@ extension NIOSSLHandler {
             promise?.fail(NIOTLSUnwrappingError.alreadyClosed)
         }
     }
+
+    /// `UnwrapTimeoutBehaviour` controls what happens if the unwrap timeout expires.
+    ///
+    /// The unwrap timeout is used to rescue the channel when `stopTLS` is called with unresponsive peers. Specifically,
+    /// if `stopTLS` is called, the `NIOSSLHandler` will issue a TLS `CLOSE_NOTIFY` alert. Some peers may not reply to
+    /// this alert, or reply very slowly. If the configured `TLSConfiguration.shutdownTimeout` expires before the remote
+    /// peer sends its reply `CLOSE_NOTIFY`, the `NIOSSLHandler` will give up. The `UnwrapTimeoutBehaviour` controls what
+    /// exactly is done in that "give up" state.
+    public struct UnwrapTimeoutBehaviour {
+        /// If the shutdown timeout expires, an error will be reported on the user promise, the `NIOSSLHandler` will
+        /// `close` the channel, and will also remove itself from the `ChannelPipeline`
+        public static let errorAndClose = UnwrapTimeoutBehaviour(.errorAndClose)
+
+        /// If the shutdown timeout expires, an error will be reported on the user promise, and the `NIOSSLHandler` will
+        /// remove itself from the `ChannelPipeline`, but it will not remove itself from the channel.
```

Comment:
I don't understand removing the handler from the pipeline but not the channel; how does that work?

Reply:
It doesn't, that should say it will not _close_ the channel.

Reply:
That makes much more sense; I should've made the connection